core/src/al/miniaudio_device.cpp

```diff
@@ -0,0 +1,238 @@
+#define MINIAUDIO_IMPLEMENTATION
+#include <cubos/core/al/miniaudio_device.hpp>
+#include <cubos/core/log.hpp>
+#include <cubos/core/reflection/external/string.hpp>
+
+using namespace cubos::core::al;
+
+MiniaudioBuffer::MiniaudioBuffer(const void* data, size_t dataSize)
+{
+    if (ma_decoder_init_memory(data, dataSize, nullptr, &mDecoder) != MA_SUCCESS)
+    {
+        CUBOS_CRITICAL("Failed to initialize Decoder from data");
+        abort();
```

Comment:
This shouldn't be fatal
```suggestion
        CUBOS_ERROR("Failed to initialize Decoder from data")
```
You should just return nullptr in the AudioDevice when a buffer isn't created successfully.